test/system/255-auto-update.bats

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# Tests for automatically update images for containerized services
+#
+
+load helpers
+
+UNIT_DIR="/usr/lib/systemd/system"
+SNAME_FILE=$BATS_TMPDIR/services
+
+function setup() {
+    skip_if_remote "systemd tests are meaningless over remote"
+    skip_if_rootless
+
+    basic_setup
+}
+
+function teardown() {
+    while read line; do
+        if [[ "$line" =~ "podman-auto-update" ]]; then
+            echo "Stop timer: $line.timer"
+            systemctl stop $line.timer
+            systemctl disable $line.timer
+        else
+            systemctl stop $line
+        fi
+        rm -f $UNIT_DIR/$line.{service,timer}
+    done < $SNAME_FILE
+
+    rm -f $SNAME_FILE
+    basic_teardown
+}
+
+function generate_service() {
+    fake_img=$1
+    target_img=$2
+    cmd=$3
+    label=$4
+
+    cname=$(random_string)
+    echo $cname
+    run_podman tag $fake_img $target_img
+    if [[ -n "$label" ]]; then
+        label="--label io.containers.autoupdate=$label"
+    fi
+    run_podman run -d --name $cname $label $target_img $cmd
+
+    run_podman generate systemd --new $cname
+    echo "$output" > "$UNIT_DIR/container-$cname.service"
+    echo "container-$cname" >> $SNAME_FILE
+    run_podman rm -f $cname
+
+    systemctl daemon-reload
+
+    run systemctl start container-$cname
+    if [ $status -ne 0 ]; then
+        die "Error starting systemd unit $SERVICE_NAME, output: $output"
+    fi
+
+    run systemctl status container-$cname
+    if [ $status -ne 0 ]; then
+        die "Non-zero status of systemd unit $SERVICE_NAME, output: $output"
+    fi
+}
+
+# This test can fail in dev. environment because of SELinux.
+# quick fix: chcon -t container_runtime_exec_t ./bin/podman
+@test "podman auto-update - label io.containers.autoupdate=image" {
+    run_podman images
+    run generate_service $IMAGE quay.io/libpod/alpine:latest top image
+    cname=${lines[0]}
```

Comment:
Whoa..... this is super dangerous and fragile. It (undocumentedly!) relies on the `echo` on line 41. There is no mention of this dependency anywhere. A future maintainer could inadvertently change the behavior of `generate_service`, and everything would break. A better way to do this would be:
```bash
   cname=$(generate_service ...)
```
And then, in `generate_service()`: (1) clearly state, in a top-level function comment, that the function outputs container name to stdout and nothing else; (2) guarantee that the function does so; (3) add a line comment just above the `echo`, repeating that this is used by the caller.